api/git/git.go

```diff
@@ -33,6 +38,16 @@ func (service *Service) ClonePrivateRepositoryWithBasicAuth(repositoryURL, refer
 }
 
 func cloneRepository(repositoryURL, referenceName string, destination string) error {
+	customClient := &http.Client{
```

Comment:
There is no need to re-create a http client each time we clone a repository, it can be re-used. I would recommend storing a pointer to the client inside the `Service` struct (could be called `httpsCli` for clarity) and initialize the http client when building a new service via `NewService`.

Reply:
OK, I will make the change